fog/distribution/src/config.rs

```diff
@@ -10,67 +12,70 @@ use mc_connection::{
-use structopt::StructOpt;
+use std::{path::PathBuf, sync::Arc};
 
-#[derive(Clone, Debug, StructOpt)]
-#[structopt(name = "fog-distribution", about = "Generate valid fog txs.")]
+/// Configuration parameters for the fog distribution utility
+#[derive(Clone, Debug, Parser)]
+#[clap(name = "fog-distribution", about = "Generate valid fog txs.")]
 pub struct Config {
     /// Path to sample data for keys/ and ledger/
-    #[structopt(long, parse(from_os_str))]
+    #[clap(long, parse(from_os_str), env = "MC_SAMPLE_DATA_DIR")]
     pub sample_data_dir: PathBuf,
 
     /// Number of transactions to send per account
-    #[structopt(long, default_value = "-1")]
+    #[clap(long, default_value = "-1", env = "MC_NUM_TX_TO_SEND")]
     pub num_tx_to_send: isize,
 
     /// Number of inputs in the ring
-    #[structopt(long, default_value = "11")]
+    #[clap(long, default_value = "11", env = "MC_RING_SIZE")]
     pub ring_size: usize,
 
     /// Block after which to tombstone
-    #[structopt(long, default_value = "50")]
+    #[clap(long, default_value = "50", env = "MC_TOMBSTONE_BLOCK")]
     pub tombstone_block: u64,
 
-    #[structopt(long, default_value = "1")]
+    /// Number of SpendableTxOut inputs.
+    #[clap(long, default_value = "1", env = "MC_NUM_INPUTS")]
     pub num_inputs: usize,
 
     /// Ask consensus for the current block to set tombstone appropriately
-    #[structopt(long)]
+    #[clap(long, env = "MC_QUERY_CONSENSUS_FOR_CUR_BLOCK")]
     pub query_consensus_for_cur_block: bool,
 
     /// Offset into transactions to start
-    #[structopt(long, default_value = "0")]
+    #[clap(long, default_value = "0", env = "MC_START_OFFSET")]
     pub start_offset: usize,
 
     /// Num transactions per account - must set this if using start_offset
-    #[structopt(long, default_value = "0")]
+    #[clap(long, default_value = "0", env = "MC_NUM_TRANSACTIONS_PER_ACCOUNT")]
     pub num_transactions_per_account: usize,
 
     /// Offset into accounts
-    #[structopt(long, default_value = "0")]
+    #[clap(long, default_value = "0", env = "MC_ACCOUNT_OFFSET")]
     pub account_offset: usize,
 
     /// Number of threads with which to submit transactions (threadpool uses min
     /// with cpu)
-    #[structopt(long, default_value = "32")]
+    #[clap(long, default_value = "32", env = "MC_MAX_THREADS")]
     pub max_threads: usize,
 
     /// Delay (in milliseconds) before each add_transaction call
-    #[structopt(long, default_value = "0")]
+    #[clap(long, default_value = "0", env = "MC_ADD_TX_DELAY_MS")]
     pub add_tx_delay_ms: u64,
 
     /// URLs to use for transaction data.
     ///
     /// For example: https://s3-us-west-1.amazonaws.com/mobilecoin.chain/node1.master.mobilecoin.com/
-    #[structopt(long, default_value = "fog_keys")]
+    #[clap(long, default_value = "fog_keys", env = "MC_FOG_KEYS_SUBDIR")]
     pub fog_keys_subdir: String,
 
-    #[structopt(flatten)]
+    /// Peers.
+    #[clap(flatten)]
```

Comment:
It's a nested structure that is inlined into this config. 

[That struct defines one flag, `peers`, which is renamed to `--peer`](https://github.com/mobilecoinfoundation/mobilecoin/blob/fbf5cdcd25e0203a7b0c93ed72d9fe14463125bb/mobilecoind/src/config.rs#L266). It can be specified multiple times, or passed a CSV of URIs

Reply:
I actually ended up inlining this particular reference, since the definition in `mobilecoind` referenced another flag that is unique to that binary.